File: Templates/Projects/C#/MainWindow.cs
using Xwt;

namespace ${Namespace}
{
	public class MainWindow : Window
	{
		public MainWindow ()
		{
			Title = "${ProjectName}";
			Width = 400;
			Height = 400;

			Menu menu = new Menu ();

			var file = new MenuItem ("_File");
			file.SubMenu = new Menu ();
			file.SubMenu.Items.Add (new MenuItem ("_Open"));
			file.SubMenu.Items.Add (new MenuItem ("_New"));
			MenuItem mi = new MenuItem ("_Close");
			mi.Clicked += (sender, e) => Close ();
			file.SubMenu.Items.Add (mi);
			menu.Items.Add (file);
			MainMenu = menu;

			var sampleLabel = new Label ("${ProjectName}");
			Content = sampleLabel;
		}

		protected override bool OnCloseRequested ()
		{
			var allow_close = MessageDialog.Confirm ("${ProjectName} will be closed", Command.Ok);
			if (allow_close)
				Application.Exit ();
			return allow_close;
		}
	}
}

File: Templates/Projects/VB/MainWindow.vb

Option Explicit On
Option Strict On

Imports System
Imports Xwt

Public Class MainWindow
	Inherits Window
	
	Public Sub New()
		Title = "${ProjectName}"
		Width = 400
		Height = 400

        MainMenu = New Menu()

        Dim FileMenu As New MenuItem("_File")
        FileMenu.SubMenu = New Menu()
        FileMenu.SubMenu.Items.Add(New MenuItem("_Open"))
        FileMenu.SubMenu.Items.Add(New MenuItem("_New"))
        Dim CloseItem As New MenuItem("_Close")
        AddHandler CloseItem.Clicked, AddressOf CloseClicked
        FileMenu.SubMenu.Items.Add(CloseItem)
        MainMenu.Items.Add(FileMenu)

        Content = New Label("${ProjectName}")
	End Sub

	Sub CloseClicked(ByVal sender As Object, ByVal e As EventArgs)
		Close ()
	End Sub

    Protected Overrides Function OnCloseRequested() As Boolean
        Dim allow_close As Boolean
        allow_close = MessageDialog.Confirm("${ProjectName} will be closed", Command.Ok)
        If allow_close Then
            Application.Exit()
        End If
        Return allow_close
    End Function
End Class
